lib/gui/pages/settings/controllers/settings.js

```diff
@@ -80,10 +86,37 @@ module.exports = function(WarningModalService, SettingsModel) {
 
     WarningModalService.display(options).then((userAccepted) => {
       if (userAccepted) {
+        AnalyticsService.logEvent('Set setting', {
+          setting: name,
+          value: true,
+          dangerous: true
+        });
+
         this.model.set(name, true);
         this.refreshSettings();
       }
     });
   };
 
+  /**
+   * @summary Log event and delegate to model.
+   * @function
+   * @public
+   *
+   * @description
+   * The logging is done here for now and not in the model to avoid
+   * a circular dependency
+   *
+   * @param {String} key - setting key
+   * @param {Object} value - setting value
+   */
+  this.set = (key, value) => {
+    AnalyticsService.logEvent('Set setting', {
+      setting: key,
+      value
```

Comment:
You're not at all miles off.

In ES6
```javascript
const x = 42;
const o = { x }
```
is just short-hand for
```javascript
const x = 42;
const o = { x: x }
```

Now that I think about it, any guide-styles for short-hand properties, @jviotti? I think at least I should group them at the beginning, as in https://github.com/airbnb/javascript#objects--grouped-shorthand